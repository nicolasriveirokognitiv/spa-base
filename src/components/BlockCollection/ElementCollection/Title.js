import React from 'react';
import { toolPropTypes } from 'Constants/propTypes';
import { Input } from 'antd';
import { elementCategories } from 'Constants/global';
import useTestMode from 'Hooks/useTestMode';

function Title({ label, onChange, value, defaultValue, testId }) {
    const testProps = useTestMode(testId);
    const handleChange = e => onChange(e.target.value);

    return (
        <Input
            placeholder={label}
            defaultValue={defaultValue}
            value={value}
            onChange={handleChange}
            {...testProps}
        />
    );
}

Title.propTypes = toolPropTypes;
Title.defaultProps = {
    label       : 'Title',
    propertyKey : 'title',
    category    : elementCategories.CONTENT,
    onChange    : () => {},
};

export default Title;
