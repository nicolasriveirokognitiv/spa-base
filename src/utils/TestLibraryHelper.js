/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
