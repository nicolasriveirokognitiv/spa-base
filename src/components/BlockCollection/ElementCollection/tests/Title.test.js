import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import Title from '../Title';

describe('Title', () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    const renderTitle = props => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <Title {...props} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<Title />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Title />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Title should be recibed on props', () => {
        const { getByTestId } = renderTitle({ value: 'Title', testId: 'title-input' });
        const elem = getByTestId('title-input');
        expect(elem.value).toBe('Title');
    });
});
