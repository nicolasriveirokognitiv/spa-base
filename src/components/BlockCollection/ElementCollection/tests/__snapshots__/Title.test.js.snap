// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="ant-input"
    placeholder="Title"
    type="text"
    value=""
  />
</DocumentFragment>
`;
